src/tests/ftest/util/nvme_utils.py

```diff
@@ -0,0 +1,451 @@
+
+        return nvme_data
+
+    def get_nvme_readlink(self):
+        """Get NVMe readlink from servers.
+
+        Args:
+            None
+
+        Returns:
+            dict: Dictionary of server readlink pci mapping with disk ID
+                  'wolf-A': {'0000:da:00.0': 'nvme9n1'}.
+                  Dictionary of server mapping with disk ID and size
+                  'wolf-A': {'nvme2n1': '1600321314816'}.
+        """
+        nvme_lsblk = self.get_nvme_lsblk()
+        nvme_readlink = {}
+
+        #Create the dictionary for NVMe readlink.
+        for server, items in nvme_lsblk.items():
+            tmp_dict = {}
+            for drive in items:
+                cmd = ('readlink /sys/block/{}/device/device'
+                       .format(drive.split()[0]))
+                task = run_task([server], cmd)
+                for _rc_code, _node in task.iter_retcodes():
+                    if _rc_code == 1:
+                        print("Failed to readlink on {}".format(_node))
+                        raise ValueError
+                #Get the drive size from each daos_io_servers
+                for buf, _node in task.iter_buffers():
+                    output = str(buf).split('\n')
+                tmp_dict[output[0].split('/')[-1]] = drive.split()[0]
+            nvme_readlink[server] = tmp_dict
+
+        return nvme_lsblk, nvme_readlink
+
+    def get_server_capacity(self):
+        """Get Server Pool NVMe storage capacity.
+
+        Args:
+            None
+
+        Returns:
+            int: Maximum NVMe storage capacity for pool creation.
+
+        Note: Read the drive sizes from the server using lsblk command.
+        This need to be replaced with dmg command when it's available.
+        This is time consuming and not a final solution to get the maximum
+        capacity of servers.
+        """
+        drive_info = {}
+        nvme_lsblk, nvme_readlink = self.get_nvme_readlink()
+
+        #Create the dictionary for NVMe size for all the servers and drives.
+        for server in nvme_lsblk:
+            tmp_dict = {}
+            for daos_io_server in range(len(self.daos_io_servers)):
+                tmp_disk_list = []
+                for disk in (self.server_managers[0].manager.job.yaml.
+                             server_params[daos_io_server].bdev_list.value):
+                    if disk in nvme_readlink[server].keys():
+                        size = int(nvme_lsblk[server]
+                                   [nvme_readlink[server][disk]])
+                        tmp_disk_list.append(size)
+                    else:
+                        self.fail("Disk {} can not found on server {}"
+                                  .format(disk, server))
+                tmp_dict[daos_io_server] = tmp_disk_list
+            drive_info[server] = tmp_dict
+
+        return self.get_max_capacity(drive_info)
+
+    def start_ior_thread(self, results, operation='WriteRead'):
```

Comment:
As before, why do we need to re-define the entire ior_cmd object and similar processed carried out as part of IorTestBase().run_ior() method? I think if we only update the ior params as needed and use run_ior() method , it should suffice the requirement. But feel free to let me know if I am missing out on something.